Jenkinsfile

```diff
@@ -7,41 +7,66 @@ pipeline {
+                        script {
+                            catchError(buildResult: 'UNSTABLE', stageResult: 'UNSTABLE') {
+                                sh "qsub -v DB build_master_index"
+                            }
+                            if (currentBuild.result == 'UNSTABLE') {
+                                // Check if the job hit the walltime. Mark the build as successful if that is the case.
+                                timeout = """${sh(returnStdout: true, script: 'grep -c "PBS: job killed: walltime [0-9]* exceeded limit" log')}""".trim()
+                                if ("$timeout"  == "1") {
+                                    echo "Database update timed out."
+                                    currentBuild.result == 'SUCCESS'
+                                } else {
+                                    error "Error updating database."
+                                }
+                            }
                         }
+
+                        sh """chmod 440 ${DB}
+                              chgrp ik11 ${DB}"""
                     }
                 }
+                stage('Check permissions') {
+                    steps {
+                        sh "./check_permissions"
+                        script {
+                            catchError(buildResult: 'UNSTABLE', stageResult: 'UNSTABLE') {
+                                // We do the next check outside the bash script, so that we can catch the error and mark the build as unstable.
+                                n_errors =  """${sh(returnStdout: true, script: 'wc -l incorrect_permissions')}""".trim()
+                                if ("$n_errors"  != "0") {
+                                    emailext (
+                                        subject: "Incorrect file permissions when indexing the COSIMA Cookbook database",
+                                        to: '${FILE, path="blame_list"}',
+                                        replyTo: 'micael.oliveira@anu.edu.au, andrew.kiss@anu.edu.au',
+                                        body: 'Dear user,\n\n While updating the COSIMA Cookbook database, we found that several files that you own have incorrect permissions or they belong to the wrong group, which prevented us from adding them to the database. You can find the full list of files below. We would be grateful if you could fix this issue by making sure that the group has read permissions for all files and directories, that the group has execute permissions for all directories, and that all files and directories belong to one of the following groups: ik11, hh5, or cj50.\n\Note that this email is generated automatically. If you believe you have received this email by mistake, please contact Micael Oliveira (micael.oliveira@anu.edu.au) or Andrew Kiss (andrew.kiss@anu.edu.au).\n\nThe following files and/or directories have incorrect permissions:\n\n${FILE, path="incorrect_permissions"}',
```

Comment:
So all users will be sent a list of all files with incorrect permissions, regardless of which they might be responsible for?

Reply:
The listing is `-ld` so I guess it will show what belongs to who. I think I would be inclined to add/change the sentence 
> You can find the full list of files below ...
to something like
> You can find the full list of files with incorrect permissions below (Note: It may be that you own only a subset of these files). We would be grateful if you could fix any issues with files owned by you by making sure that ...

Reply:
I think that's a very good suggestion. I'll change the code.